Dockerfile

```diff
@@ -95,6 +91,15 @@ RUN \
     && echo 'Cleaning up installation files' >&2 && \
     apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
+# install required python packages from file
+COPY pydeps.txt /tmp/pydeps.txt
```

Comment:
Maybe it was unclear (or voluntary not taken in to account) but these files should be called <sthg>_requirements.txt. As stated in the [pip documentation](https://pip.readthedocs.io/en/latest/user_guide/#requirements-files).

Moreover, they should be defined in the RIOT repository and not the docker file. This will obviously help new developers setting-up their development environment!

Reply:
no in practice you wouldn't build the container image yourself, you pull it from docker hub - which we support with this repo. See https://hub.docker.com/r/riot/riotbuild/

Reply:
> this allows for testing and publishing the container

This is also the case with a Dockerfile in RIOT by the way.

> makes reviewing such things independent of the RIOT reviewing process

What is wrong with the RIOT reviewing process ? too strict ? ok

Reply:
> This is also the case with a Dockerfile in RIOT by the way.

how? Murdock won't try to build the Dockerfile and why should it? Here we have a distinct travis config that serves the task of building - maybe we should add some testing, too.

> What is wrong with the RIOT reviewing process ? too strict ? ok

no, but it's meant for code review ... not tooling and config - and with nearly 500 open PRs for mostly code stuff, things like this will likely get lost, delayed and forgotten.

---

divide and conquer, remember 😉

Reply:
@aabadie so you do want to have this discussion here :)

Currently, there's some limited devops deployment strategy with the riotdocker repo. Every commit to master triggers a Docker hub rebuild, which when successful triggers a Murdock worker container rebuild, which if successful will then automatically be picked up by the workers eventually.

Docker hub rebuilds on push. We don't want that on every push to RIOT master.
Can we theoretically think of (script-fu or whatever) ways around that? Of course. Is it worth the hassle? I don't think so.

Having the Dockerfile in a seperate repository makes handling it much less complex. As long as there's no pressing reason to keep the Docker stuff in-sync with the main repository, that is just the sane way to go.